sentry/src/main/java/io/sentry/IScopes.java

```diff
@@ -0,0 +1,594 @@
+   * transaction options the decision if transaction is sampled will be taken by {@link
+   * TracesSampler}.
+   *
+   * @param transactionContext the transaction context
+   * @param transactionOptions the transaction options
+   * @return created transaction.
+   */
+  @NotNull
+  ITransaction startTransaction(
+      final @NotNull TransactionContext transactionContext,
+      final @NotNull TransactionOptions transactionOptions);
+
+  /**
+   * Returns the "sentry-trace" header that allows tracing across services. Can also be used in
+   * &lt;meta&gt; HTML tags. Also see {@link IHub#getBaggage()}.
+   *
+   * @deprecated please use {@link IHub#getTraceparent()} instead.
+   * @return sentry trace header or null
+   */
+  @Deprecated
+  @Nullable
+  SentryTraceHeader traceHeaders();
+
+  /**
+   * Associates {@link ISpan} and the transaction name with the {@link Throwable}. Used to determine
+   * in which trace the exception has been thrown in framework integrations.
+   *
+   * @param throwable the throwable
+   * @param span the span context
+   * @param transactionName the transaction name
+   */
+  @ApiStatus.Internal
+  void setSpanContext(
+      @NotNull Throwable throwable, @NotNull ISpan span, @NotNull String transactionName);
+
+  /**
+   * Gets the current active transaction or span.
+   *
+   * @return the active span or null when no active transaction is running
+   */
+  @Nullable
+  ISpan getSpan();
+
+  /**
+   * Returns the transaction.
+   *
+   * @return the transaction or null when no active transaction is running.
+   */
+  @ApiStatus.Internal
+  @Nullable
+  ITransaction getTransaction();
+
+  /**
+   * Gets the {@link SentryOptions} attached to current scope.
+   *
+   * @return the options attached to current scope.
+   */
+  @NotNull
+  SentryOptions getOptions();
+
+  /**
+   * Returns if the App has crashed (Process has terminated) during the last run. It only returns
+   * true or false if offline caching {{@link SentryOptions#getCacheDirPath()} } is set with a valid
+   * dir.
+   *
+   * <p>If the call to this method is early in the App lifecycle and the SDK could not check if the
+   * App has crashed in the background, the check is gonna do IO in the calling thread.
+   *
+   * @return true if App has crashed, false otherwise, and null if not evaluated yet
+   */
+  @Nullable
+  Boolean isCrashedLastRun();
+
+  /**
+   * Report a screen has been fully loaded. That means all data needed by the UI was loaded. If
+   * time-to-full-display tracing {{@link SentryOptions#isEnableTimeToFullDisplayTracing()} } is
+   * disabled this call is ignored.
+   *
+   * <p>This method is safe to be called multiple times. If the time-to-full-display span is already
+   * finished, this call will be ignored.
+   */
+  void reportFullyDisplayed();
+
+  /**
+   * @deprecated See {@link IHub#reportFullyDisplayed()}.
+   */
+  @Deprecated
+  default void reportFullDisplayed() {
```

Comment:
Given that we'll do a major bump I guess it's worth considering removing deprecated methods.

Reply:
Created https://github.com/getsentry/sentry-java/issues/3364 to track and added it to `8.0.0` milestone.